### PR TITLE
[Fix] [Server] Fix the spell error and add log in SqlTask

### PR DESCRIPTION
**Pull request by zixi0825**
## What is the purpose of the pull request

*This pull request fix the spell error 、add log in SqlTask and add null judgment in SqlTask*

## Brief change log

  - *none*

## Verify this pull request

*none*

### CODE REVIEW

**Comment by yangyichao-mango** [comment]
LGTM.

**Comment by zhuangchong** [comment]
1. SqlParameters = = null, because the key parameters have been checked in the UI, where SqlParameters cannot be null

2. For the exception of connection closing, the log should be added. In my opinion, if an exception occurs, the connection = = null
---
1. sqlParameters ==null , 因为在UI里面已经对关键参数进行校验，此处sqlParameters不可能为null
2.针对connection关闭发生异常增加日志，我认为更应该如果发生异常将connection==null

**Comment by yangyichao-mango** [comment]
> 1. SqlParameters = = null, because the key parameters have been checked in the UI, where SqlParameters cannot be null
> 2. For the exception of connection closing, the log should be added. In my opinion, if an exception occurs, the connection = = null
> 
> 1. sqlParameters ==null , 因为在UI里面已经对关键参数进行校验，此处sqlParameters不可能为null
>    2.针对connection关闭发生异常增加日志，我认为更应该如果发生异常将connection==null

Hi,
1. In addition to using webui, users can also use API to create directly, and it can also solve the null value caused by errors in the JSON serialization process. Judging null value here actually can enhance the robustness of the code.
Example: Although the value of a `String` cannot be null, we usually use the `StringUtils.isNotEmpty`, and it add the null value judgment.
2. Please point out where the code is.

-----
1.用户除了使用webui，也可以使用api直接创建，除此之外可以将将json序列化过程中出错导致的null值给解决。这里判断是否为null，可以增强代码鲁棒性。举例：虽然一个string的值不可能为null，我们通常使用的StringUtils.isNotEmpty中依然加了null值判断。
2.可以指出具体是哪段代码嘛。

If you have any question or suggestion, welcome to put forward~
Thx a lot~

**Comment by zhuangchong** [comment]
@yangyichao-mango

1. I haven't used the method of creating API directly. It's not clear. I see that the method of checkprocessnodelist in dolphinscheduler-api  module can also detect parameters. I don't know if this method will be used to create API directly. You can have a look.   

```
public static boolean checkTaskNodeParameters(String parameter, String taskType) {
    AbstractParameters abstractParameters = TaskParametersUtils.getParameters(taskType, parameter);

    if (abstractParameters != null) {
      return abstractParameters.checkParameters();
    }

    return false;
  }
```

2. This PR is to add an exception log in sqltask
```
	try {
                resultSet.close();	               
            } catch (SQLException e) {	           

 +              logger.error("close result set error :",e);
            }
```
I think more should be added resultSet==null,pstmt==null, connection==null
```
	try {
                resultSet.close();	               
            } catch (SQLException e) {	           
 +		 resultSet == null;
 +              logger.error("close result set error :",e);
            }	 
 ``` 
---
1. 你说的api直接创建这种方式我没有用过，这个不太清楚，我看dolphinscheduler-api 模块里面有checkProcessNodeList方法也会检测参数。不知道你说的api直接创建会不会走这个方法，你可以看一下。
```
public static boolean checkTaskNodeParameters(String parameter, String taskType) {
    AbstractParameters abstractParameters = TaskParametersUtils.getParameters(taskType, parameter);

    if (abstractParameters != null) {
      return abstractParameters.checkParameters();
    }

    return false;
  }
```
2. 本次PR是在SqlTask中增加 异常日志
```
	try {
                resultSet.close();	               
            } catch (SQLException e) {	           

 +              logger.error("close result set error :",e);
            }
```
我认为更应该增加的是resultSet==null,pstmt==null, connection==null
```
	try {
                resultSet.close();	               
            } catch (SQLException e) {	           
 +		 resultSet == null;
 +              logger.error("close result set error :",e);
            }	 
 ```

**Comment by zhuangchong** [comment]
If SqlParameters is judged to be null, are all abstractparameters required? If so, they need to be abstracted
---
如果SqlParameters判断为空，那么是不是所有的 AbstractParameters的都需要，如果是，就需要抽象出来

**Comment by yangyichao-mango** [comment]
> ## If SqlParameters is judged to be null, are all abstractparameters required? If so, they need to be abstracted
> 如果SqlParameters判断为空，那么是不是所有的 AbstractParameters的都需要，如果是，就需要抽象出来

I agree with you, if you are interested in this, I think it is better to create an issue and describe all the situation. Thx a lot~

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ public SqlTask(TaskExecutionContext taskExecutionContext, Logger logger) {
         logger.info("sql task params {}", taskExecutionContext.getTaskParams());
         this.sqlParameters = JSONUtils.parseObject(taskExecutionContext.getTaskParams(), SqlParameters.class);
 
-        if (!sqlParameters.checkParameters()) {
+        if (sqlParameters == null || (!sqlParameters.checkParameters())) {
```
Comment (gabrywu) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java:90]: why this condition removed?

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java [hunk 2]
```diff
@@ -367,33 +367,33 @@ private void close(ResultSet resultSet,
             try {
                 resultSet.close();
             } catch (SQLException e) {
-
+                logger.error("close result set error :",e);
```
Comment (gabrywu) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java:370]: logger.error("close result set error: {}",e.getMessage(),e)    better

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java (hunk 1 above)
Comment (zixi0825) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java:90]: Which condition was removed? I just added non-empty judgment

Comment (zhuangchong) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java:90]: Non empty verification has been carried out in the API -module，checkTaskNodeParameters  method
---
API 模块 里面已经进行非空验证，checkTaskNodeParameters 方法

**Pull request merged**